distributed/comm/tcp.py

```diff
@@ -299,16 +299,25 @@ async def write(self, msg, serializers=None, on_error="message"):
             # trick to enque all frames for writing beforehand
             for each_frame_nbytes, each_frame in zip(frames_nbytes, frames):
                 if each_frame_nbytes:
-                    if stream._write_buffer is None:
-                        raise StreamClosedError()
+                    each_frame = memoryview(each_frame)
 
                     if isinstance(each_frame, memoryview):
                         # Make sure that `len(data) == data.nbytes`
                         # See <https://github.com/tornadoweb/tornado/pull/2996>
                         each_frame = ensure_memoryview(each_frame)
 
```

Comment:
This can now be dropped with the `memoryview` coercion above since this is handled there

```suggestion

```